jwst/residual_fringe/residual_fringe.py

```diff
@@ -83,11 +83,13 @@ def do_correction(self):
             raise ErrorNoFringeFlat("The fringe flat step has not been run on file %s",
                                     self.input_model.meta.filename)
 
-        # Remove any NaN values from the data prior to processing
+        # Remove any NaN values and flagged DO_NOT_USE pixels from the data prior to processing
         # Set them to 0 for the residual fringe routine
         # They will be re-added at the end
         output_data = self.model.data.copy()
-        nanval_indx = np.where(~np.isfinite(output_data))
+        DO_NOT_USE = datamodels.dqflags.pixel["DO_NOT_USE"]
+        nanval_indx = np.where(np.logical_and(np.bitwise_and(self.model.dq, DO_NOT_USE).astype(bool),
```

Comment:
I think this needs to be np.logical_or instead if the aim is to catch anything that has non-finite SCI values or DO_NOT_USE set.  In an ideal world those would be the same, but it doesn't hurt to check both.

Reply:
Goodness good thing you saw that.  That is the point of the entire PR. 
I do not think they are the same from a regression test I just did. Looking in more detail at the results.